plugins/faustwp/includes/utilities/callbacks.php

```diff
@@ -31,27 +32,29 @@
  *
  * @link https://developer.wordpress.org/reference/functions/register_activation_hook/
  *
+ * @param bool $network_active True if plugin is network activated.
+ *
  * @return void
  */
-function handle_activation() {
+function handle_activation( $network_active ) {
 	if ( is_plugin_active( 'wpe-headless/wpe-headless.php' ) ) {
 		deactivate_plugins( 'wpe-headless/wpe-headless.php', true );
 	}
 
-	$secret_key = get_secret_key();
-	$settings   = faustwp_get_settings();
+	// Handle network activation of plugin within multisite.
+	if ( $network_active ) {
+		$sites = get_sites();
```

Comment:
maybe here check if the function get_sites exists:
```php
function_exists( 'get_sites' )
```

Reply:
or maybe as well place a limit:
```
$sites = get_sites( [
				'fields' => 'ids',
				'number' => PHP_INT_MAX,
			] );
```

Reply:
@theodesp Thanks for the feedback!

I didn't include
```
function_exists( 'get_sites' )
```
as [get_sites()](https://developer.wordpress.org/reference/functions/get_sites/) was introduced WordPress 4.6 & [faustwp requires a minimum WP version of 5.7](https://github.com/wpengine/faustjs/blob/canary/plugins/faustwp/readme.txt#L4).

Good call on only selecting ids, I can add that in! In regards to explicitly setting a limit for get_sites(), WP defaults to 100; just want to clarify you're suggesting we remove that limit?

Reply:
We can also take advantage of [wp_is_large_network()](https://developer.wordpress.org/reference/functions/wp_is_large_network/) if needed.

Reply:
Yes maybe we should consider both cases here I didn't know that.